saphir/src/server.rs

```diff
@@ -53,6 +53,31 @@ static INIT_STACK: Once = Once::new();
 #[doc(hidden)]
 static REQUEST_FUTURE_COUNT: AtomicU64 = AtomicU64::new(0);
 
+struct StackAlreadyInitialized;
+
+fn write_into_static(stack: Stack, server_value: HeaderValue, request_body_max: Option<usize>) -> Result<&'static Stack, StackAlreadyInitialized> {
```

Comment:
Since the "Error" type `StackAlreadyInitialized` is not being used by either invocation of `write_into_static` (see line 557 & 485), this function should return an `Result<&'static Stack, SaphirError>`, while `StackAlreadyInitialized ` should become a variant of that type. This way you won't have to map these errors, and it will prevent cluttering up the code with unused error types.